File: support_tools/swigwin-3.0.12/Examples/test-suite/php5/char_strings_runme.php
<?php

require "tests.php";
require "char_strings.php";

$CPLUSPLUS_MSG = "A message from the deep dark world of C++, where anything is possible.";
$OTHERLAND_MSG_10 = "Little message from the safe world.10";

check::equal(GetCharHeapString(), $CPLUSPLUS_MSG, "failed GetCharHeapString");
check::equal(GetConstCharProgramCodeString(), $CPLUSPLUS_MSG, "failed GetConstCharProgramCodeString");
check::equal(GetCharStaticString(), $CPLUSPLUS_MSG, "failed GetCharStaticString");
check::equal(GetCharStaticStringFixed(), $CPLUSPLUS_MSG, "failed GetCharStaticStringFixed");
check::equal(GetConstCharStaticStringFixed(), $CPLUSPLUS_MSG, "failed GetConstCharStaticStringFixed");

check::equal(SetCharHeapString($OTHERLAND_MSG_10, 10), true, "failed GetConstCharStaticStringFixed");
check::equal(SetCharStaticString($OTHERLAND_MSG_10, 10), true, "failed SetCharStaticString");
check::equal(SetCharArrayStaticString($OTHERLAND_MSG_10, 10), true, "failed SetCharArrayStaticString");
check::equal(SetConstCharHeapString($OTHERLAND_MSG_10, 10), true, "failed SetConstCharHeapString");
check::equal(SetConstCharStaticString($OTHERLAND_MSG_10, 10), true, "failed SetConstCharStaticString");
check::equal(SetConstCharArrayStaticString($OTHERLAND_MSG_10, 10), true, "failed SetConstCharArrayStaticString");

check::equal(CharPingPong($OTHERLAND_MSG_10), $OTHERLAND_MSG_10, "failed CharPingPong");

Global_char_set($OTHERLAND_MSG_10);
check::equal(Global_char_get(), $OTHERLAND_MSG_10, "failed Global_char_get");

Global_char_array1_set($OTHERLAND_MSG_10);
check::equal(Global_char_array1_get(), $OTHERLAND_MSG_10, "failed Global_char_array1_get");

Global_char_array2_set($OTHERLAND_MSG_10);
check::equal(Global_char_array2_get(), $OTHERLAND_MSG_10, "failed Global_char_array2_get");

check::equal(Global_const_char_get(), $CPLUSPLUS_MSG, "failed Global_const_char");
check::equal(Global_const_char_array1_get(), $CPLUSPLUS_MSG, "failed Global_const_char_array1");
check::equal(Global_const_char_array2_get(), $CPLUSPLUS_MSG, "failed Global_const_char_array2");

check::equal(GetCharPointerRef(), $CPLUSPLUS_MSG, "failed GetCharPointerRef");
check::equal(SetCharPointerRef($OTHERLAND_MSG_10, 10), true, "failed SetCharPointerRef");
check::equal(GetConstCharPointerRef(), $CPLUSPLUS_MSG, "failed GetConstCharPointerRef");
check::equal(SetConstCharPointerRef($OTHERLAND_MSG_10, 10), true, "failed SetConstCharPointerRef");

check::done();
?>

File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/char_strings_runme.cs

using System;
using char_stringsNamespace;

public class char_strings_runme {

  private static string CPLUSPLUS_MSG = "A message from the deep dark world of C++, where anything is possible.";
  private static string OTHERLAND_MSG = "Little message from the safe world.";

  public static void Main() {

    uint count = 10000;
    uint i = 0;

    // get functions
    for (i=0; i<count; i++) {
      string str = char_strings.GetCharHeapString();
      if (str != CPLUSPLUS_MSG)
        throw new Exception("Test char get 1 failed, iteration " + i);
      char_strings.DeleteCharHeapString();
    }

    for (i=0; i<count; i++) {
      string str = char_strings.GetConstCharProgramCodeString();
      if (str != CPLUSPLUS_MSG)
        throw new Exception("Test char get 2 failed, iteration " + i);
      char_strings.DeleteCharHeapString();
    }

    for (i=0; i<count; i++) {
      string str = char_strings.GetCharStaticString();
      if (str != CPLUSPLUS_MSG)
        throw new Exception("Test char get 3 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      string str = char_strings.GetCharStaticStringFixed();
      if (str != CPLUSPLUS_MSG)
        throw new Exception("Test char get 4 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      string str = char_strings.GetConstCharStaticStringFixed();
      if (str != CPLUSPLUS_MSG)
        throw new Exception("Test char get 5 failed, iteration " + i);
    }

    // set functions
    for (i=0; i<count; i++) {
      if (!char_strings.SetCharHeapString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 1 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetCharStaticString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 2 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetCharArrayStaticString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 3 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetConstCharHeapString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 4 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetConstCharStaticString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 5 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetConstCharArrayStaticString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 6 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetCharConstStaticString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 7 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetConstCharConstStaticString(OTHERLAND_MSG + i, i))
        throw new Exception("Test char set 8 failed, iteration " + i);
    }

    // get set function
    for (i=0; i<count*10; i++) {
      string ping = OTHERLAND_MSG + i;
      string pong = char_strings.CharPingPong(ping);
      if (ping != pong)
        throw new Exception("Test PingPong 1 failed.\nExpected:" + ping + "\nReceived:" + pong);
    }

    // variables
    for (i=0; i<count; i++) {
      char_strings.global_char = OTHERLAND_MSG + i;
      if (char_strings.global_char != OTHERLAND_MSG + i)
        throw new Exception("Test variables 1 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      char_strings.global_char_array1 = OTHERLAND_MSG + i;
      if (char_strings.global_char_array1 != OTHERLAND_MSG + i)
        throw new Exception("Test variables 2 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      char_strings.global_char_array2 = OTHERLAND_MSG + i;
      if (char_strings.global_char_array2 != OTHERLAND_MSG + i)
        throw new Exception("Test variables 3 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (char_strings.global_const_char != CPLUSPLUS_MSG)
        throw new Exception("Test variables 4 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (char_strings.global_const_char_array1 != CPLUSPLUS_MSG)
        throw new Exception("Test variables 5 failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (char_strings.global_const_char_array2 != CPLUSPLUS_MSG)
        throw new Exception("Test variables 6 failed, iteration " + i);
    }

    // char *& tests
    for (i=0; i<count; i++) {
      String str = char_strings.GetCharPointerRef();
      if (str != CPLUSPLUS_MSG)
        throw new Exception("Test char pointer ref get failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetCharPointerRef(OTHERLAND_MSG + i, i))
        throw new Exception("Test char pointer ref set failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      String str = char_strings.GetConstCharPointerRef();
      if (str != CPLUSPLUS_MSG)
        throw new Exception("Test const char pointer ref get failed, iteration " + i);
    }

    for (i=0; i<count; i++) {
      if (!char_strings.SetConstCharPointerRef(OTHERLAND_MSG + i, i))
        throw new Exception("Test const char pointer ref set failed, iteration " + i);
    }
  }
}

